devtools/db_import/db_import/config.py

```diff
@@ -4,4 +4,55 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+import pathlib
+import yaml
+import contextlib
+import os
+
+from typing import TextIO
+
 PIPELINES_KEY: str = "pipelines"
+
+
+@contextlib.contextmanager
+def _current_working_directory(path: pathlib.Path):
+  previous_cwd = pathlib.Path().absolute()
+
+  try:
+    os.chdir(path)
+    yield path
+  finally:
+    os.chdir(previous_cwd)
+
+
+def _embed(loader: yaml.Loader, node):
+  """This tag handler allows loading the contents of file as string into a node.
+
+    The initial node's content is interpreted as a file path.
+    Relative file path are expected to be relative to the YAML file's location.
+    Absolute paths are also allowed but must refer to a location below the YAML
+    file's parent directory.
+
+    Example:
+    ```yaml
+    ---
+    property: !embed filename.txt
+    """
+  yaml_filepath = pathlib.Path(loader.name).resolve()
+
+  with _current_working_directory(yaml_filepath.parent):
```

Comment:
I'm thinking if we should cd to a resolved path. This seems to cause an issue when the `yaml_filepath.parent` is or under a symlink but the `filepath = pathlib.Path(loader.construct_scalar(node)).resolve()` will resolve the symlink.

Reply:
Yes. Great catch. Changed.